src/Datadog.Trace.ClrProfiler.Native/il_rewriter.cpp

```diff
@@ -341,13 +348,14 @@ ILInstr* ILRewriter::NewILInstr() {
   return new ILInstr();
 }
 
-ILInstr* ILRewriter::GetInstrFromOffset(unsigned offset) {
-  ILInstr* pInstr = NULL;
+HRESULT ILRewriter::GetInstrFromOffset(unsigned offset, ILInstr** ppInstr) {
+  *ppInstr = nullptr;
```

Comment:
This seems a little dangerous. Instead of assigning to the input/output pointer here, could this be a local `ILInstr*`. And then the input/output pointer could be assigned only if the operation succeeded?

Reply:
Thanks. I refactored this in 33ea6ce219170a33f31aaac61ce458acecd79e65.